ProductApp/src/main/java/com/stack/jspdemo/model/Product.java

```diff
@@ -0,0 +1,127 @@
+import javax.persistence.NamedStoredProcedureQuery;
+import javax.persistence.ParameterMode;
+import javax.persistence.StoredProcedureParameter;
+import javax.validation.constraints.Size;
+
+@Entity
+@NamedStoredProcedureQueries({
+		@NamedStoredProcedureQuery(name = "findAllProducts", procedureName = "getAllProducts", resultClasses = {
+				Product.class }),
+		@NamedStoredProcedureQuery(name = "insertProduct", procedureName = "insertProd", parameters = {
+				@StoredProcedureParameter(name = "productName", type = String.class, mode = ParameterMode.IN),
+				@StoredProcedureParameter(name = "model", type = String.class, mode = ParameterMode.IN),
+				@StoredProcedureParameter(name = "price", type = String.class, mode = ParameterMode.IN), }) })
+public class Product {
+
+	@Id
+	@GeneratedValue
+	private Integer id;
+	@Size(min = 3, max = 45, message = "Name should be at leats 3 characters")
+	private String name;
+	private String model;
+	private String price;
+
+	public Product() {
+
+	}
+
+	public Product(Integer id, String name, String model, String price) {
+		super();
+		this.id = id;
+		this.name = name;
+		this.model = model;
+		this.price = price;
+	}
+
+	public Integer getId() {
+		return id;
+	}
+
+	public void setId(Integer id) {
+		this.id = id;
+	}
+
+	public Product(String name) {
+		super();
+		this.name = name;
+	}
+
+	public String getName() {
+		return name;
+	}
+
+	public void setName(String name) {
+		this.name = name;
+	}
+
+	public String getModel() {
+		return model;
+	}
+
+	public void setModel(String model) {
+		this.model = model;
+	}
+
+	public String getPrice() {
+		return price;
+	}
+
+	public void setPrice(String price) {
+		this.price = price;
+	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((id == null) ? 0 : id.hashCode());
+		result = prime * result + ((model == null) ? 0 : model.hashCode());
+		result = prime * result + ((name == null) ? 0 : name.hashCode());
+		result = prime * result + ((price == null) ? 0 : price.hashCode());
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		Product other = (Product) obj;
+		if (id == null) {
+			if (other.id != null)
+				return false;
+		} else if (!id.equals(other.id))
+			return false;
+		if (model == null) {
+			if (other.model != null)
+				return false;
+		} else if (!model.equals(other.model))
+			return false;
+		if (name == null) {
+			if (other.name != null)
+				return false;
+		} else if (!name.equals(other.name))
+			return false;
+		if (price == null) {
+			if (other.price != null)
+				return false;
+		} else if (!price.equals(other.price))
+			return false;
+		return true;
+	}
+
+	@Override
+	public String toString() {
+		return "Product [id=" + id + ", name=" + name + ", model=" + model + ", price=" + price + "]";
```

Comment:
Why not using `String.format()` instead of concatenating the string like this?

Reply:
done